x-pack/plugins/security_solution/public/common/components/charts/donutchart.tsx

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiFlexGroup, EuiFlexItem, EuiLink, EuiText, EuiTextColor } from '@elastic/eui';
+import React, { useContext, useMemo } from 'react';
+
+import { Chart, Datum, Partition, Settings, PartitionLayout } from '@elastic/charts';
+import uuid from 'uuid';
+import styled from 'styled-components';
+import { ThemeContext } from './donut_theme_context';
+import { useTheme } from './common';
+import { DraggableLegend } from './draggable_legend';
+import { LegendItem } from './draggable_legend_item';
+import { escapeDataProviderId } from '../drag_and_drop/helpers';
+import { DonutChartEmpty } from './donutchart_empty';
+import { ParsedSeverityBucket } from '../../../overview/components/detection_response/alerts_by_status/types';
+
+export const NO_LEGEND_DATA: LegendItem[] = [];
+
+type DonutChartData = ParsedSeverityBucket;
```

Comment:
We can not tight a _common_ component implementation to one particular use case. We are doing that same assumption at lines 126 and 132 with `groupByRollup: (d: Datum) => d.status,`.

We have to keep this component agnostic, forcing a common data type. Or otherwise, we can also move this donut chart component to our specific alerts_by_status folder.